### ISSUE TITLE
Class variables sometimes cannot be referenced

### ISSUE THREAD

**Issue by viraptor** [python/mypy#740]
Referencing a class variable in context of list comprehension doesn't work. This code raises `class_var.py:3: error: Name 'some_var' is not defined`

```
class X():
    some_var = [1]
    other_var = [a for a in some_var]
```

However dropping the list (`other_var = some_var`) passes without issues.

**Comment by JukkaL** [comment]
Verified that I can reproduce the bug. Thanks for reporting this!

**Comment by sanxiyn** [comment]
This is very funny. This works:

```python
class C:
    a = [1]
    b = [x for x in a]
```

But this doesn't!

```python
class C:
    a = [1]
    b = [(x, y) for x in a for y in a]
```

The reason is that the outermost for-expression is special and evaluated in a different scope. This is very explicitly stated in [PEP 289](https://www.python.org/dev/peps/pep-0289/) with rationale, but most people find it surprising.

**Comment by JelleZijlstra** [comment]
Fixed as of 0.600 (and probably earlier).
